src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,122 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_LENGTH = 6;
+    private T[] array = (T[]) new Object[DEFAULT_ARRAY_LENGTH];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        resize();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAddByIndex(index);
+        resize();
 
+        if (index < size) {
+            System.arraycopy(array, index, array, index + 1, array.length - index - 1);
+        }
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return removeElementFromList(index);
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index;
+        try {
+            index = findElementForRemove(element);
+        } catch (NoSuchMethodException e) {
+            throw new NoSuchElementException();
+        }
+        return removeElementFromList(index);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private T removeElementFromList(int index) {
+        T result = array[index];
+        if (index == size - 1) {
+            size--;
```

Comment:
Is that enough to delete an object? Also assign a null to the item with the right index